### PR TITLE
Docs - Template walk through, update getting started

### PR DESCRIPTION
**Pull request by karikushi**
- update the Getting Started guide so that version requirements are clearer and add how to configure the app
- create a walk-through creating a new app using the Template as a base

### CODE REVIEW

File: docs/examples/app-the-new-thing/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Change Log
+
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/CHANGELOG.md]: Why is this file here? `app-the-new-thing` hasn't been changed yet so this change log is misleading and unnecessary.

File: docs/examples/app-the-new-thing/interbit.prod.config.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const path = require('path')
+const chainAliases = require('./src/constants/chainAliases')
+
+const PUBLIC_KEY =
+  '-----BEGIN PGP PUBLIC KEY BLOCK-----\r\nVersion: OpenPGP.js v2.6.2\r\nComment: https://openpgpjs.org\r\n\r\nxk0EWtpPiAEB/1DUOOu08SW7IGGlw5AavcxUxtrJbJVliIcFNSTpn/z/p0Zi\nIfO58AK0dfcHyMb1vUY8zwM45if6iaNS98zF3lEAEQEAAc0NPGluZm9AYnRs\nLmNvPsJ1BBABCAApBQJa2k+IBgsJBwgDAgkQjFLIxmtXVSMEFQgKAgMWAgEC\nGQECGwMCHgEAAIdvAf0SbWcBMphrR7wc6rL5ytyThLBsI72vz/0QyBcaRlsp\nQ9US66w6f+OWcpAiOeLDdx9l39difSXpjL9yYWxWRElSzk0EWtpPiAECAOpL\nfIIdC5S/lIaWI+Bx23FtSdxyqrKduDQCRDhB07udTv4bjGCSCtpyPS3Y03m6\nyl/GAa7OLIFeLzI4tzT0CXMAEQEAAcJfBBgBCAATBQJa2k+ICRCMUsjGa1dV\nIwIbDAAAxXwB/RUA88XTd6vDJDFeRx4/Escv5tyQuT9bxMkmSxaqiBRTU2X5\nhrFQs5NGOu2ySGbRvZMopK91sLK/uqlTaty1oVk=\r\n=yws5\r\n-----END PGP PUBLIC KEY BLOCK-----\r\n\r\n'
+
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/interbit.prod.config.js]: Keys again

File: docs/examples/app-the-new-thing/interbit.prod.config.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const config = {
+  // TODO: Ensure that at least one account chain is set as a peer
+  peers: ['ib-dev----master.herokuapp.com:443'],
+  adminValidators: [PUBLIC_KEY],
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/interbit.prod.config.js]: Peer without context. It may be better to bind to process.env.PROD_PEERS or similar rather than hard coding it.

File: docs/examples/app-the-new-thing/interbit.prod.config.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    'template-control': {
+      location: path.join(__dirname, 'src/interbit/control')
+    },
+    'template-private': {
+      location: path.join(__dirname, 'src/interbit/private')
+    }
+  },
+  apps: {
+    template: {
+      peers: ['ib-dev----master.herokuapp.com'], // the peers the browser should connect to
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/interbit.prod.config.js]: Peers again. How about process.env.PROD_APP_PEERS or similar.

File: docs/examples/app-the-new-thing/interbit.prod.config.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    // dynamically created chains
+    'template-public': {
+      location: path.join(__dirname, 'src/interbit/public')
+    },
+    'template-control': {
+      location: path.join(__dirname, 'src/interbit/control')
+    },
+    'template-private': {
+      location: path.join(__dirname, 'src/interbit/private')
+    }
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/interbit.prod.config.js]: Please rename the covenant aliases or they will conflict with template if we try to deploy them.

File: docs/examples/app-the-new-thing/interbit.config.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    'template-public': {
+      location: path.join(__dirname, 'src/interbit/public')
+    },
+    'template-control': {
+      location: path.join(__dirname, 'src/interbit/control')
+    },
+    'template-private': {
+      location: path.join(__dirname, 'src/interbit/private')
+    }
+  },
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/interbit.config.js]: Please rename the covenant aliases.

File: docs/examples/app-the-new-thing/interbit.config.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  apps: {
+    theNewThing: {
+      peers: ['localhost:5000'], // the peers the browser should connect to
+      chains: [chainAliases.PUBLIC, chainAliases.CONTROL], // the chains that need to load in the browser
+      appChain: chainAliases.PUBLIC, // The chain that the static page is loaded on
+      indexLocation: path.join(__dirname, 'public/index.html'), // the index.html to update with the app info
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/interbit.config.js]: I haven't seen the chainAliases file yet. I assume/hope you have changed the aliases so they won't conflict with template. Please and thank you.

File: docs/examples/app-the-new-thing/public/index.html [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+  <script src="https://unpkg.com/interbit-core@0.8.2/dist/bundle.browser.js"></script>
+  <title>Interbit Application Template</title>
+</head>
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/public/index.html]: Change the title to `New Application based on Interbit Application Template`

File: docs/examples/app-the-new-thing/src/index.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+const sagaMiddleware = createSagaMiddleware()
+
+const store = createStore(
+  reducers,
+  composeWithDevTools(applyMiddleware(interbitMiddleware, sagaMiddleware))
+)
+sagaMiddleware.run(interbitSaga)
+
+// BlockExplorer will monitor the public chain
+store.dispatch(setSelectedChain(CONTROL))
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/src/index.js]: Comment is wrong. Just remove it.

File: docs/examples/app-the-new-thing/src/interbit/control/constants.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  CHAIN_ALIASES: {
+    CONTROL: 'templateControl',
+    PUBLIC: 'templatePublic',
+    PRIVATE: 'templatePrivate'
+  },
+  COVENANTS: {
+    PRIVATE: 'template-private'
+  }
+}
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/src/interbit/control/constants.js]: Please change chain and covenant aliases to avoid naming collisions. This is IMPORTANT. Make sure it is documented too.

File: docs/examples/app-the-new-thing/src/interbit/control/index.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    payload: {
+      whoMeowed
+    }
+  })
+}
+
+const initialState = Immutable.from({
+  // Not required but handy for troubleshooting
+  chainMetadata: { chainName: `Template application - control chain` },
+
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/src/interbit/control/index.js]: It's no longer the template application.

File: docs/examples/app-the-new-thing/src/interbit/control/index.js [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+          process.env.INTERBIT_PROJECTS_URL || 'http://localhost:3035'
+      }
+    }
+  },
+
+  // Information required for creating private chains using the
+  // Public/Private/Control (PPC) model
+  privateChainHosting: {},
+  cats: {}
+})
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/src/interbit/control/index.js]: Add a comment for cats. This is new chain state added to the control chain, not part of the PPC model.

**** METADATA LEAKAGE ***** THIS NEEDS TO BE FIXED ASAP!

I want to stop this repurposing of the control chain NOW, before users do it. We should have a 'shared' chain in the template specifically for this purpose to avoid polluting the PPC model. Anyone who loads the control chain will see every private chain ID for every user, will know how many use accounts have been created, etc., etc. That is a real bad thing!

File: docs/examples/app-the-new-thing/src/interbit/control/index.js [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+        actionPermissions: {
+          [actionTypes.MEOW]: 'coolcats'
+        },
+        roles: { coolcats: [privateChainPublicKey] }
+      })
+
+      return redispatch(nextState, addToAclAction)
+    }
+
+    case actionTypes.MEOW: {
```
Comment (djlmills) [inline docs/examples/app-the-new-thing/src/interbit/control/index.js]: Put this in a shared chain. Not in the control chain.

File: docs/examples/app-the-new-thing/interbit.prod.config.js (hunk 3 above)
Comment (nicholeous) [reply docs/examples/app-the-new-thing/interbit.prod.config.js]: I think I'm just going to nix the prod config for this example

File: docs/examples/app-the-new-thing/interbit.config.js [hunk 14]
```diff
@@ -0,0 +1,82 @@
+const path = require('path')
+const chainAliases = require('./src/constants/chainAliases')
+
+const PUBLIC_KEY =
+  '-----BEGIN PGP PUBLIC KEY BLOCK-----\r\nVersion: OpenPGP.js v2.6.2\r\nComment: https://openpgpjs.org\r\n\r\nxk0EWtpPiAEB/1DUOOu08SW7IGGlw5AavcxUxtrJbJVliIcFNSTpn/z/p0Zi\nIfO58AK0dfcHyMb1vUY8zwM45if6iaNS98zF3lEAEQEAAc0NPGluZm9AYnRs\nLmNvPsJ1BBABCAApBQJa2k+IBgsJBwgDAgkQjFLIxmtXVSMEFQgKAgMWAgEC\nGQECGwMCHgEAAIdvAf0SbWcBMphrR7wc6rL5ytyThLBsI72vz/0QyBcaRlsp\nQ9US66w6f+OWcpAiOeLDdx9l39difSXpjL9yYWxWRElSzk0EWtpPiAECAOpL\nfIIdC5S/lIaWI+Bx23FtSdxyqrKduDQCRDhB07udTv4bjGCSCtpyPS3Y03m6\nyl/GAa7OLIFeLzI4tzT0CXMAEQEAAcJfBBgBCAATBQJa2k+ICRCMUsjGa1dV\nIwIbDAAAxXwB/RUA88XTd6vDJDFeRx4/Escv5tyQuT9bxMkmSxaqiBRTU2X5\nhrFQs5NGOu2ySGbRvZMopK91sLK/uqlTaty1oVk=\r\n=yws5\r\n-----END PGP PUBLIC KEY BLOCK-----\r\n\r\n'
+
```
[the remark this answers is not in the document]
Comment (nicholeous) [reply docs/examples/app-the-new-thing/interbit.config.js]: Good catch. We should probably show the users how to change these in the getting started guide before they hit trouble and move them there.

File: docs/examples/app-the-new-thing/README.md [hunk 15]
```diff
@@ -0,0 +1,2310 @@
+This project was bootstrapped with [Create React App](https://github.com/facebookincubator/create-react-app).
```
[the remark this answers is not in the document]
Comment (nicholeous) [reply docs/examples/app-the-new-thing/README.md]: 👌

File: docs/examples/app-the-new-thing/interbit.prod.config.js (hunk 4 above)
Comment (nicholeous) [reply docs/examples/app-the-new-thing/interbit.prod.config.js]: I'm not sure we should use env vars in a walkthrough as it adds extra steps but we ought to find another way to eliminate this

**Comment by djlmills** [comment]
On reflection (dog walking) I have had a chance to think about my primary objection to this, the repurposing of the control chain. The PPC model includes Public, Private and Control chains. What you are doing on the Control chain actually belongs on the Public Chain. So the chain responsibilities are:

*Private Chain*: Each user has their own and only the user can load and interact with it.

*Public Chain*: Every user can load and interact with this chain. It contains public data and MAY contain functionality that all users can interact with by dispatching actions.

*Control Chain*: This is the administrative chain that manages the creation of private chains and is the source of truth of system data on the public chain (shared via a read join). Only administrators have access to this chain and it should NEVER be loaded on users browsers.

The whole point of the public chain is that it is available to every user of the system. The misconception here seems to be caused by the fact that the template does not dispatch actions to the public chain.

We should add some covenant actions to the public chain so that all users can interact with it and modify the control chain covenant to dispatch actions to the public chain to add users public keys to the public chain (so users can dispatch directly to the public chain) or configure write joins between the users private chains and the public chain (so users can dispatch actions to their private chain and have them forwarded to the public chain as in the UTH sample).

**Comment by nicholeous** [comment]
Per PR #242 and standup conversations we will break this walkthrough into at least 3 different privacy architectures with corresponding walkthroughs.

Closing this PR then re-opening a simplified template walkthrough.
